### PR TITLE
A honcharenko/multiarch build 801

### PR DESCRIPTION
**Pull request by a-honcharenko**
## What
Build and publish docker multiarch images to have native arm64/amd64 images.
Modified github runner AMI.  

## Recommended reading order
1. ./tools/bin/release_version.sh
2. ./tools/bin/publish_docker.sh

### CODE REVIEW

**Review by supertopher — CHANGES_REQUESTED** [review]
requesting changes because I think you are already making changes and I wanted that clearer in my notifications.  Ping me if I'm mistaken please

**Comment by a-honcharenko** [comment]
Looks like it's working now. I returned bump_version stuff that I disabled to check publishing. 
@supertopher @davinchia  please review changes when you'll have a time. If you have any suggestions or comments, please let me know. 
Thanks!

File: airbyte-workers/Dockerfile
Comment (supertopher) [inline airbyte-workers/Dockerfile:15]: I haven't looked at the output from `arch`, but do you intend to replace `aarch` not `arch` here?

```suggestion
RUN arch=$(arch | sed s/arch64/arm64/ | sed s/x86_64/amd64/) && \
```

File: tools/bin/publish_docker.sh
Comment (supertopher) [inline tools/bin/publish_docker.sh:20]: this syntax is always a pain for me.  Can you verify that you tested these locally?
when I try it, the default does not stick.  this output is from my command line
```
bash-3.2$ POSTGRES_IMAGE2=${POSTGRES_IMAGE2:-postgres:13-alpine} echo postgres2 is $POSTGRES_IMAGE2
postgres2 is
```

File: tools/bin/publish_docker.sh
Comment (supertopher) [inline tools/bin/publish_docker.sh:33]: nit: what is the purpose of the `&&` here? because we `set -e` above `&&` will be seen as a test and therefore a valid command flow even with an exit 1.  This will cause lines 33 to ignore `set -e` (as opposed to exiting like we usually expect `set -e` to do)

This is a shell oddity so I made a test script to test my assumption.  I just want to make sure you intend to continue execution if line 33 exits 1

test script
```
runbooks $ cat test.sh                                                                                    
set -e

invalid_command && echo cool
echo i still run

actual_error
echo i dont run
runbooks $ bash test.sh                                                                                      
test.sh: line 3: invalid_command: command not found
i still run
test.sh: line 6: actual_error: command not found
```

File: tools/bin/publish_docker.sh
Comment (supertopher) [inline tools/bin/publish_docker.sh:40]: beautiful formatting 😍

**Review by supertopher — COMMENTED** [review]
looks good overall, just a few questions

File: .github/actions/start-aws-runner/action.yml
Comment (davinchia) [inline .github/actions/start-aws-runner/action.yml:12]: curious, what does the new ami contain?

File: airbyte-container-orchestrator/Dockerfile
Comment (davinchia) [inline airbyte-container-orchestrator/Dockerfile:15]: can we leave a comment here to explain why we need this?

File: airbyte-workers/Dockerfile
Comment (davinchia) [inline airbyte-workers/Dockerfile:16]: same comment here on explaining what this for

File: tools/bin/release_version.sh
Comment (davinchia) [inline tools/bin/release_version.sh:31]: can we add a comment here explaining what this container is for?

File: tools/bin/publish_docker.sh
Comment (a-honcharenko) [reply tools/bin/publish_docker.sh:20]: Yes, I tested it locally. It works as expected. When you have env var, it overrides default value in script.

File: tools/bin/publish_docker.sh
Comment (a-honcharenko) [reply tools/bin/publish_docker.sh:33]: In this case it executes in a loop. It should work as I expected because if any command will return !=0, script will fail and execution will brake in current iteration.

File: .github/actions/start-aws-runner/action.yml
Comment (a-honcharenko) [reply .github/actions/start-aws-runner/action.yml:12]: Added buildx binary and enabled features.

File: tools/bin/publish_docker.sh
Comment (a-honcharenko) [reply tools/bin/publish_docker.sh:33]: And one more thing. 
Line `docker buildx rm $artifactName` should be executed anyway, because if you'll have created context with the same name you'll have to remove it manually.

**Review by davinchia — APPROVED** [review]
Nicely done! Most of my comments are about adding more comments to future readers.

I double checked this builds locally, and it did. We should kick off one release after the merge to make sure this works.

Want to wait for Topher/Peter to give the green light before merging.

File: airbyte-workers/Dockerfile
Comment (a-honcharenko) [reply airbyte-workers/Dockerfile:15]: aarch64 are using arm64 in this case. Debian repo for aarch64 don't exists.

File: tools/bin/publish_docker.sh
Comment (supertopher) [reply tools/bin/publish_docker.sh:20]: I know the override works, it's the default that I'm worried about ☝️

File: .github/actions/start-aws-runner/action.yml
Comment (git-phu) [inline .github/actions/start-aws-runner/action.yml:12]: just for my own understanding, how do we build new images?
For example do we use packer or something similar, or do we just spin up a new ubuntu instance and make changes to that, or something else?

In general is there some way for us to see what changes are included in these images (other than asking whoever created the PR)?

**Review by git-phu — APPROVED** [review]
looks good!

just had one question

File: tools/bin/publish_docker.sh
Comment (supertopher) [reply tools/bin/publish_docker.sh:20]: awesome, thanks @git-phu

File: .github/actions/start-aws-runner/action.yml
Comment (a-honcharenko) [reply .github/actions/start-aws-runner/action.yml:12]: For now we are spin up a new ubuntu instance and make changes.

There is no way now to view changes that was made on AMI. I think packer is a good solution to do that, but it requires investigation. We need all changes that was applied on this image or create it from scratch.

Comment (git-phu) [reply .github/actions/start-aws-runner/action.yml:12]: got it, thanks!

Comment (davinchia) [reply .github/actions/start-aws-runner/action.yml:12]: @git-phu https://github.com/airbytehq/airbyte-cloud/blob/master/docs/Things-To-Know/Builds/Builds.md

Comment (davinchia) [reply .github/actions/start-aws-runner/action.yml:12]: on the packer front - it's a trade off between making it accessible to everyone and having to bring on something more specialised. maybe packer is worth it.

Comment (git-phu) [reply .github/actions/start-aws-runner/action.yml:12]: agreed.  I think for me the value to something like packer is it essentially documents what is in each ami.  Otherwise, I'm not sure we have a good way to keep track of that at current.

Comment (davinchia) [reply .github/actions/start-aws-runner/action.yml:12]: an interesting thought I had is what if we used docker instead of packer. it's much more accessible vs packer. but maybe there are too many downside. we can discuss somewhere else!

**Review by supertopher — APPROVED** [review]

**Pull request merged**
